app/controllers/declaration/SummaryController.scala

```diff
@@ -64,23 +69,39 @@ class SummaryController @Inject()(
   }
 
   val submitDeclaration: Action[AnyContent] = (authenticate andThen verifyEmail andThen journeyType).async { implicit request =>
-    LegalDeclaration.form.bindFromRequest
-      .fold(
+    def submit(form: Form[LegalDeclaration]): Future[Result] =
+      form.bindFromRequest.fold(
         (formWithErrors: Form[LegalDeclaration]) => Future.successful(BadRequest(normalSummaryPage(formWithErrors))),
         legalDeclaration => {
           submissionService.submit(request.eori, request.cacheModel, legalDeclaration).map {
             case Some(submission) =>
               Redirect(routes.ConfirmationController.displayHoldingPage).withSession(session(submission))
-
             case _ => handleError(s"Error from Customs Declarations API")
           }
         }
       )
+
+    isLrnADuplicate(request.cacheModel.lrn match {
+      case Some(lrn) => Some(Lrn(lrn))
+      case _         => None
+    }).flatMap {
```

Comment:
To aid code readability try to avoid performing logic in calls to methods, instead you can do the logic just above the method and assign it to a var and then pass that var to the method, e.g,
```
val maybeLrn = request.cacheModel.lrn.map(Lrn(_))

isLrnADuplicate(maybeLrn).flatMap { ...

``` 
:wink: